python_ta/__init__.py

```diff
@@ -346,24 +355,25 @@ def get_file_paths(rel_path: AnyStr) -> Generator[AnyStr, None, None]:
                 yield os.path.join(root, filename)  # Format path, from root.
 
 
-def _verify_pre_check(filepath: AnyStr) -> bool:
+def _verify_pre_check(filepath: AnyStr, linter: PyLinter) -> bool:
     """Check student code for certain issues."""
     # Make sure the program doesn't crash for students.
     # Could use some improvement for better logging and error reporting.
     try:
-        # Check for inline "pylint:" comment, which may indicate a student
-        # trying to disable a check.
-        with tokenize.open(os.path.expanduser(filepath)) as f:
-            for tok_type, content, _, _, _ in tokenize.generate_tokens(f.readline):
-                if tok_type != tokenize.COMMENT:
-                    continue
-                match = OPTION_PO.search(content)
-                if match is not None:
-                    print(
-                        '[ERROR] String "pylint:" found in comment. '
-                        + "No check run on file `{}.`\n".format(filepath)
-                    )
-                    return False
+        if not linter.config.allow_pylint_comments:
```

Comment:
This is a case where an *early return* at the top of this function is more appropriate; it acts as a *guard* for the rest of the code. This would look like

```python
if linter.config.allow_pylint_comments:
    return True

...  # Rest of code is unchanged
```

This will minimize the amount of changes you're making in this code.